lib/server.js

```diff
@@ -889,73 +885,18 @@ Server.prototype._run = function _run(req, res, route, chain, cb) {
-                        self.emit('FormatterError', req, res, route, arg);
-                    } else {
-                        // otherwise, log it and send empty response.
-                        log.error(arg, 'error formatting response, ' +
-                                       'sending empty payload!');
-                        res.end('');
-                    }
-                    // return early.
-                    return self._finishReqResCycle(req, res, route, arg);
-                }
-
-                var errName = arg.name.replace(/Error$/, '');
-                log.trace({
-                    err: arg,
-                    errName: errName
-                }, 'next(err=%s)', (arg.name || 'Error'));
-
-                // always attempt to use the most specific error listener
-                // possible. fall back on generic 'error' listener if we can't
-                // find one for the error we got.
-                var hasErrListeners = false;
-                var errEvtNames = [];
-
-                // if we have listeners for the specific error
-                if (self.listeners(errName).length > 0) {
-                    hasErrListeners = true;
-                    errEvtNames.push(errName);
-                }
-                // or if we have a generic error listener
-                if (self.listeners('restifyError').length > 0) {
-                    hasErrListeners = true;
-                    errEvtNames.push('restifyError');
-                }
-
-                if (hasErrListeners) {
-                    vasync.forEachPipeline({
-                        func: function emitError(errEvtName, vasyncCb) {
-                            self.emit(errEvtName, req, res, arg,
-                            function errEvtHandlerDone() {
-                                // the error listener may return arbitrary
-                                // objects, throw them away and continue on.
-                                // don't want vasync to take that error and
-                                // stop.
-                                return vasyncCb();
-                            });
-                        },
-                        inputs: errEvtNames
-                    }, function (err, results) {
-                        res.send(err || arg);
-                        return (cb ? cb(err || arg) : true);
-                    });
-                    emittedError = true;
-                } else {
+                // the emitting of the error events are async, so we can not
+                // complete this invocation of run() until it returns. setting
+                // the emittedError flag is flag that is returned sync -
+                // setting this flag will allow us to to not call cb()
+                // automatically later in this function.
+                emittedError = self._emitErrorEvents(req, res, route, arg,
+                function emitErrorsDone() {
```

Comment:
Yep! Lint doesn't complain because function invocations don't have to be indented even if they are > 1 line.

I actually prefer this style, as it gives us an extra 4 spaces for this last arg. Open to suggestions on how to better do this and still keep within 80cols though.